security-utils/src/test/java/com/yahoo/security/tls/authz/PeerAuthorizerTest.java

```diff
@@ -0,0 +1,140 @@
+import java.math.BigInteger;
+import java.security.KeyPair;
+import java.security.cert.X509Certificate;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Set;
+
+import static com.yahoo.security.SignatureAlgorithm.SHA256_WITH_ECDSA;
+import static com.yahoo.security.tls.policy.RequiredPeerCredential.Field.CN;
+import static com.yahoo.security.tls.policy.RequiredPeerCredential.Field.SAN_DNS;
+import static java.util.Collections.emptySet;
+import static java.util.stream.Collectors.toSet;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author bjorncs
+ */
+public class PeerAuthorizerTest {
+
+    private static final KeyPair KEY_PAIR = KeyUtils.generateKeypair(KeyAlgorithm.EC);
+
+    @Test
+    public void certificate_must_match_both_san_and_cn_pattern() {
+        String roleName = "role";
+        String policyName = "policy";
+        RequiredPeerCredential cnRequirement = createRequiredCredential(CN, "*.matching.cn");
+        RequiredPeerCredential sanRequirement = createRequiredCredential(SAN_DNS, "*.matching.san");
+        PeerAuthorizer authorizer = createPeerAuthorizer(createPolicy(policyName, createRoles(roleName), cnRequirement, sanRequirement));
+
+        AuthorizationResult result = authorizer.authorizePeer(createCertificate("foo.matching.cn", "foo.matching.san", "foo.invalid.san"));
+        assertAuthorized(result);
+        assertEquals(1, result.assumedRoles().size());
+        assertEquals(roleName, result.assumedRoles().iterator().next().name());
+        assertEquals(1, result.matchedPolicies().size());
+        assertEquals(policyName, result.matchedPolicies().iterator().next());
+
+        assertUnauthorized(authorizer.authorizePeer(createCertificate("foo.invalid.cn", "foo.matching.san")));
+        assertUnauthorized(authorizer.authorizePeer(createCertificate("foo.invalid.cn", "foo.matching.san", "foo.invalid.san")));
+        assertUnauthorized(authorizer.authorizePeer(createCertificate("foo.matching.cn", "foo.invalid.san")));
+    }
+
+    @Test
+    public void can_match_multiple_policies() {
+        RequiredPeerCredential cnRequirement = createRequiredCredential(CN, "*.matching.cn");
+        RequiredPeerCredential sanRequirement = createRequiredCredential(SAN_DNS, "*.matching.san");
+        PeerAuthorizer peerAuthorizer = createPeerAuthorizer(
+                createPolicy("policy-1", createRoles("role-1", "role-2"), cnRequirement, sanRequirement),
+                createPolicy("policy-2", createRoles("role-2", "role-3"), cnRequirement, sanRequirement));
+
+        AuthorizationResult result = peerAuthorizer
+                .authorizePeer(createCertificate("foo.matching.cn", "foo.matching.san"));
+        assertAuthorized(result);
+        assertEquals(3, result.assumedRoles().size());
+        assertEquals(2, result.matchedPolicies().size());
```

Comment:
Checking the sets themselves would be nice, e.g. with a `containsInAnyOrder` matcher to make things more readable